Document custom slugs for content collection entries (#2214)
withastro/docs: src/pages/en/guides/content-collections.md

📚 See [Zod’s documentation](https://github.com/colinhacks/zod) for a complete list of features.

> ### Custom entry slugs

Reviewer: ```suggestion
## Custom entry slugs
```
Suggesting this as a separate entry in `defineCollections` and not a part of defining a schema

> By default, Astro will generate a `slug` for each content entry based on its file path. If you want to generate custom slugs for each entry, you can provide a `slug()` function in `defineCollection()`. A custom `slug()` function will be passed the entry ID, default slug, parsed frontmatter (as `data`), and the raw body of the entry.

Author: Took a pass at this. _You_ don’t pass those things, Astro does. It’s like “here’s what you get to work with when figuring out a good ID”. I’ve tried to reflect that in the updated text.

```ts {5-9}
// src/content/config.ts

> import { defineCollection } from 'astro:content';

Reviewer: ```suggestion
import { defineCollection, z } from 'astro:content';
```
I think because a schema is shown, we'd need `z` imported here too, right?

const blog = defineCollection({
  slug: ({ id, defaultSlug, data, body }) => {

>     // If set, use `permalink` from the entry’s frontmatter as the slug.

Reviewer: ```suggestion
    // Use `permalink` from the entry’s frontmatter as the slug, if it exists.
```
The "if set" threw me off at first. ("If a slug function is set?") I think maybe it's clearer to start with the intention here.

Would also tend to go with "if it exists" or "if found", but that might just be a preference thing.
